gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineReleaseCompatibility.java

```diff
@@ -53,17 +53,23 @@ public void apply(Project project) {
     private static final class ReleaseFlagProvider implements CommandLineArgumentProvider {
         private final JavaCompile javaCompile;
 
+        private final boolean hasBaselineJavaVersionPlugin;
+        private final String projectString;
+
         private ReleaseFlagProvider(JavaCompile javaCompile) {
             this.javaCompile = javaCompile;
+            Project project = javaCompile.getProject();
+            this.hasBaselineJavaVersionPlugin = project.getPlugins().hasPlugin(BaselineJavaVersion.class);
```

Comment:
I don't believe this is safe to do. When you create this object - which is during configuration - the java version plugin could potentially be applied *after* this plugin. Meaning the `hasBaselineJavaVersionPlugin` is actually wrong, and behaviour is subtly changed. It still needs to happen as late as possible.